static/app/components/notificationActions/notificationActionManager.tsx

```diff
@@ -0,0 +1,203 @@
+import {Fragment, useState} from 'react';
+import capitalize from 'lodash/capitalize';
+
+import DropdownButton from 'sentry/components/dropdownButton';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import NotificationActionItem from 'sentry/components/notificationActions/notificationActionItem';
+import {IconAdd} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Project} from 'sentry/types';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+  NotificationActionService,
+} from 'sentry/types/notificationActions';
+
+type NotificationActionManagerProps = {
+  /**
+   * The list of existing notification actions
+   */
+  actions: NotificationAction[];
+  /**
+   * The list of available notification actions
+   */
+  availableActions: AvailableNotificationAction[];
+  /**
+   * The project associated with the notification actions
+   * TODO(enterprise): refactor to account for multiple projects
+   */
+  project: Project;
+  /**
+   * Updates the notification alert count for this project
+   */
+  updateAlertCount: (projectId: number, alertCount: number) => void;
+  /**
+   * Optional list of roles to display as recipients of Sentry notifications
+   */
+  recipientRoles?: string[];
+};
+
+const NotificationActionManager = ({
+  actions,
+  availableActions,
+  recipientRoles,
+  project,
+  updateAlertCount,
+}: NotificationActionManagerProps) => {
+  const [notificationActions, setNotificationActions] =
+    useState<Partial<NotificationAction>[]>(actions);
+
+  const addNotificationAction = (
```

Comment:
🤔 

does this need to be a separate helper?
I see it's only used once below, can we not in-line this?